aexpect/client.py

```diff
@@ -522,6 +529,7 @@ def __init__(self, command=None, a_id=None, auto_close=False, echo=False,
         :param encoding: Override text encoding (by default: autodetect by
                 locale.getpreferredencoding())
         """
+        self.tail_thread = None
```

Comment:
This change is valid but not relevant to this fix. We ought to set the `self.tail_thread` before adding the `Tail._join_thread` to prevent it's usage on cleanup. Could you please use a separate commit with explanation for this change?

Reply:
I add this because when using the Python terminal, even if the ExpectProcessTerminatedError will be raised, another exception will also be raised when the Python terminal closes.
```python
>>> from aexpect import client
>>> process = client.run_tail("ping localhost")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python3.9/site-packages/aexpect/client.py", line 1491, in run_tail
    bg_process = Tail(command=command,
  File "/usr/local/lib/python3.9/site-packages/aexpect/client.py", line 538, in __init__
    super().__init__(command, a_id, auto_close, echo, linesep,
  File "/usr/local/lib/python3.9/site-packages/aexpect/client.py", line 206, in __init__
    raise ExpectProcessTerminatedError(pattern, sub_status, "")
aexpect.exceptions.ExpectProcessTerminatedError: Process terminated while looking for patterns 'Server a04jkNXg ready'    (status: 1,    output: '')
>>>
Exception ignored in: <function Spawn.__del__ at 0xffffa10cbb80>
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/aexpect/client.py", line 241, in __del__
    self.close()
  File "/usr/local/lib/python3.9/site-packages/aexpect/client.py", line 399, in close
    hook(self)
  File "/usr/local/lib/python3.9/site-packages/aexpect/client.py", line 701, in _join_thread
    thread = self.tail_thread
AttributeError: 'Tail' object has no attribute 'tail_thread'
```

I agree with your suggestion, of using a separate commit to explain this issue.